lib/spack/spack/package.py

```diff
@@ -1044,7 +1090,11 @@ def do_patch(self):
             except spack.multimethod.NoSuchMethodError:
                 # We are running a multimethod without a default case.
                 # If there's no default it means we don't need to patch.
-                tty.msg("No patches needed for %s" % self.name)
+                if not patched:
+                    # if we didn't apply a patch, AND the patch function
```

Comment:
Suggestion: "if we didn't apply a patch from a patch directive, AND..."

"apply a patch" is general and I had to read this several times to figure out what it meant.